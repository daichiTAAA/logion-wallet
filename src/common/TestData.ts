import { Coin, CoinBalance, SYMBOL } from '../logion-chain/Balances';
import { PrefixedNumber, ATTO } from '../logion-chain/numbers';

import Identity from './types/Identity';
import PostalAddress from './types/PostalAddress';
import { Transaction } from './types/ModelTypes';
import { ColorTheme, rgbaToHex } from './ColorTheme';

export const DEFAULT_LEGAL_OFFICER = "5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY"; // Alice
export const ANOTHER_LEGAL_OFFICER = "5FHneW46xGXgs5mUiveU4sbTyGBzmstUspZC92UhjJM694ty"; // Bob
export const A_THIRD_LEGAL_OFFICER = "5FLSigC9HGRKVhB9FiEo4Y3koPsNmBmLJbpXg2mp1hXcS59Y"; // Charlie

export const DEFAULT_IDENTITY: Identity = {
    firstName: "John",
    lastName: "Doe",
    email: "john.doe@logion.network",
    phoneNumber: "+1234",
};

export const DEFAULT_ADDRESS: PostalAddress = {
    line1: "Place de le République Française, 10",
    line2: "boite 15",
    postalCode: "4000",
    city: "Liège",
    country: "Belgium",
};

export const COLOR_THEME: ColorTheme = {
    type: 'light',
    shadowColor: rgbaToHex('#3b6cf4', 0.1),
    dashboard: {
        background: '#eff3fe',
        foreground: '#000000',
    },
    sidebar: {
        background: '#ffffff',
        foreground: '#000000',
        activeItemBackground: '#d8e2fd',
    },
    accounts: {
        iconBackground: '#3b6cf4',
        hintColor: rgbaToHex('#000000', 0.6),
        foreground: '#000000',
        background: 'eff3fe',
        legalOfficerIcon: {
            category: 'legal-officer',
            id: 'account-shield'
        }
    },
    frame: {
        background: '#ffffff',
        foreground: '#000000',
        link: rgbaToHex('#3b6cf4', 0.15),
        altBackground: '#eff3fe',
    },
    topMenuItems: {
        iconGradient: {
            from: '#3b6cf4',
            to: '#6050dc',
        }
    },
    bottomMenuItems: {
        iconGradient: {
            from: '#7a90cb',
            to: '#3b6cf4',
        }
    },
    recoveryItems: {
        iconGradient: {
            from: '#7a90cb',
            to: '#3b6cf4',
        }
    },
    buttons: {
        secondary: {
            background: '#eff3fe',
            foreground: '#000000',
        }
    },
    select: {
        background: '#ffffff',
        foreground: '#000000',
        menuBackgroundColor: '#ffffff',
        selectedOptionBackgroundColor: '#ffffff',
    },
    table: {
        background: '#0c163d',
        foreground: '#ffffff',
        header: {
            background: '#0c163d',
            foreground: '#ffffff',
        },
        row: {
            background: '#152665',
            foreground: '#ffffff',
        }
    },
    tabs: {
        background: '#0c163d',
        foreground: '#ffffff',
        inactiveTab: {
            background: '#d8e2fd',
            foreground: '#ffffff',
        },
        borderColor: '#3b6cf4',
    },
    dialog: {
        background: '#eff3fe',
        foreground: '#000000',
        borderColor: '#e6007a',
    },
};

export const DEFAULT_COIN: Coin = {
    id: 'log',
    iconId: 'log',
    iconType: 'png',
    name: "Logion",
    symbol: SYMBOL
};

export const DEFAULT_COIN_BALANCE: CoinBalance = {
    coin: DEFAULT_COIN,
    balance: new PrefixedNumber("42", ATTO),
    level: 0.1,
};

export const DEFAULT_TRANSACTION: Transaction = {
    from: "from",
    to: "to",
    pallet: "pallet",
    method: "method",
    transferValue: "420",
    tip: "0",
    fee: "125000",
    reserved: "0",
    total: "125420",
    createdOn: "2021-07-28T12:30:00.000",
    type: "Sent",
}
