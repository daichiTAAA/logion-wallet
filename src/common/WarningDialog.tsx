import React from 'react';

import Dialog, { ModalSize } from './Dialog';
import { Action } from './Button';
import Icon from './Icon';

import { Children } from './types/Helpers';
import { ColorTheme } from './ColorTheme';

import './WarningDialog.css';

export interface Props {
    show: boolean,
    modalTestId?: string,
    actions: Action[],
    size: ModalSize,
    colors: ColorTheme,
    children: Children,
}

export default function WarningDialog(props: Props) {

    return (
        <Dialog
            show={ props.show }
            size={ props.size }
            actions={ props.actions }
            colors={ props.colors }
            modalTestId={ props.modalTestId }
        >
            <Icon
                colorThemeType={ props.colors.type }
                icon={{
                    id: "warning"
                }}
            />
            <p className="dialog-text">
                { props.children }
            </p>
        </Dialog>
    );
}
