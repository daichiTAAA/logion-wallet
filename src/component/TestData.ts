import Identity from './types/Identity';
import PostalAddress from './types/PostalAddress';
import { ColorTheme, rgbaToHex } from '../component/ColorTheme';

export const DEFAULT_IDENTITY: Identity = {
    firstName: "John",
    lastName: "Doe",
    email: "john.doe@logion.network",
    phoneNumber: "+1234",
};

export const DEFAULT_ADDRESS: PostalAddress = {
    line1: "Place de le République Française, 10",
    line2: "boite 15",
    postalCode: "4000",
    city: "Liège",
    country: "Belgium",
};

export const COLOR_THEME: ColorTheme = {
    type: 'light',
    shadowColor: rgbaToHex('#3b6cf4', 0.1),
    dashboard: {
        background: '#eff3fe',
        foreground: '#000000',
    },
    sidebar: {
        background: '#ffffff',
        foreground: '#000000',
        activeItemBackground: '#d8e2fd',
    },
    accounts: {
        iconBackground: '#3b6cf4',
        hintColor: rgbaToHex('#000000', 0.6),
        foreground: '#000000',
        background: 'eff3fe',
        legalOfficerIcon: {
            category: 'legal-officer',
            id: 'account-shield'
        }
    },
    frame: {
        background: '#ffffff',
        foreground: '#000000',
        link: rgbaToHex('#3b6cf4', 0.15),
        altBackground: '#eff3fe',
    },
    topMenuItems: {
        iconGradient: {
            from: '#3b6cf4',
            to: '#6050dc',
        }
    },
    bottomMenuItems: {
        iconGradient: {
            from: '#7a90cb',
            to: '#3b6cf4',
        }
    },
    buttons: {
        secondary: {
            background: '#eff3fe',
            foreground: '#000000',
        }
    },
    select: {
        background: '#ffffff',
        foreground: '#000000',
        menuBackgroundColor: '#ffffff',
        selectedOptionBackgroundColor: '#ffffff',
    },
    table: {
        background: '#0c163d',
        foreground: '#ffffff',
        header: {
            background: '#0c163d',
            foreground: '#ffffff',
        },
        row: {
            background: '#152665',
            foreground: '#ffffff',
        }
    },
    tabs: {
        background: '#0c163d',
        foreground: '#ffffff',
        inactiveTab: {
            background: '#d8e2fd',
            foreground: '#ffffff',
        },
        borderColor: '#3b6cf4',
    },
    dialog: {
        background: '#eff3fe',
        foreground: '#000000',
        borderColor: '#e6007a',
    },
};
