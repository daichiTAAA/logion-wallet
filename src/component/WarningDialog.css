.Dialog .dialog-text {
    font-weight: bold;
    margin-top: 20px;
}
