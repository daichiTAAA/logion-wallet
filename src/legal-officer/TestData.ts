import moment from 'moment';
import { toIsoString } from '../logion-chain/datetime';
import { ProtectionRequest, DEFAULT_LEGAL_OFFICER } from './Types';

export const PENDING_PROTECTION_REQUESTS: ProtectionRequest[] = [
    {
        id: "1",
        requesterAddress: "5Ew3MyB15VprZrjQVkpQFj8okmc9xLDSEdNhqMMS5cXsqxoW",
        decisions: [
            {
                legalOfficerAddress: DEFAULT_LEGAL_OFFICER,
                status: 'PENDING',
                rejectReason: null,
                decisionOn: null
            }
        ],
        userIdentity: {
            firstName: "John",
            lastName: "Doe",
            email: "john.doe@logion.network",
            phoneNumber: "+1234",
        },
        userPostalAddress: {
            line1: "Place de le République Française, 10",
            line2: "boite 15",
            postalCode: "4000",
            city: "Liège",
            country: "Belgium",
        },
        createdOn: toIsoString(moment('2021-06-10T11:40:00.000Z'))
    }
];

export const PROTECTION_REQUESTS_HISTORY: ProtectionRequest[] = [
    {
        id: "1",
        requesterAddress: "5Ew3MyB15VprZrjQVkpQFj8okmc9xLDSEdNhqMMS5cXsqxoW",
        decisions: [
            {
                legalOfficerAddress: DEFAULT_LEGAL_OFFICER,
                status: 'ACCEPTED',
                rejectReason: null,
                decisionOn: toIsoString(moment('2021-06-10T11:40:00.000Z'))
            }
        ],
        userIdentity: {
            firstName: "John",
            lastName: "Doe",
            email: "john.doe@logion.network",
            phoneNumber: "+1234",
        },
        userPostalAddress: {
            line1: "Place de le République Française, 10",
            line2: "boite 15",
            postalCode: "4000",
            city: "Liège",
            country: "Belgium",
        },
        createdOn: toIsoString(moment('2021-06-10T11:40:00.000Z'))
    }
];
