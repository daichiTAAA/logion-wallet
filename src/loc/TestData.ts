import { LegalOfficerCase } from "@logion/node-api/dist/Types";
import { UUID } from "@logion/node-api/dist/UUID";

import { LocRequest } from "../common/types/ModelTypes";

export function buildLocRequest(locId: UUID, loc: LegalOfficerCase): LocRequest {
    return {
        ownerAddress: loc.owner,
        requesterAddress: loc.requesterAddress,
        requesterIdentityLoc: loc.requesterLocId?.toString(),
        description: "Description",
        locType: loc.locType,
        createdOn: "",
        decisionOn: "",
        id: locId.toString(),
        status: loc.closed ? "CLOSED" : "OPEN",
        files: loc.files.map((locFile, index) => ({
            ...locFile,
            name: `File ${index}`,
            addedOn: "",
        })),
        metadata: loc.metadata.map((locFile, index) => ({
            ...locFile,
            name: `Data ${index}`,
            addedOn: "",
        })),
        links: loc.links.map((locFile, index) => ({
            ...locFile,
            name: `Link ${index}`,
            addedOn: "",
            target: locFile.id.toString(),
        }))
    };
}
