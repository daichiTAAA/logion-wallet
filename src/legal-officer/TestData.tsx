import moment from 'moment';
import { toIsoString } from '@logion/node-api/dist/datetime';
import { ProtectionRequest } from '@logion/client/dist/RecoveryClient';

import { DEFAULT_LEGAL_OFFICER, ANOTHER_LEGAL_OFFICER } from "../common/TestData";

export const PENDING_PROTECTION_REQUESTS: ProtectionRequest[] = [
    {
        id: "1",
        requesterAddress: "5Ew3MyB15VprZrjQVkpQFj8okmc9xLDSEdNhqMMS5cXsqxoW",
        legalOfficerAddress: DEFAULT_LEGAL_OFFICER,
        otherLegalOfficerAddress: ANOTHER_LEGAL_OFFICER,
        decision: {
            rejectReason: null,
            decisionOn: null,
        },
        userIdentity: {
            firstName: "John",
            lastName: "Doe",
            email: "john.doe@logion.network",
            phoneNumber: "+1234",
        },
        userPostalAddress: {
            line1: "Place de le République Française, 10",
            line2: "boite 15",
            postalCode: "4000",
            city: "Liège",
            country: "Belgium",
        },
        createdOn: toIsoString(moment('2021-06-10T11:40:00.000Z')),
        isRecovery: false,
        addressToRecover: null,
        status: "PENDING"
    }
];

export const PROTECTION_REQUESTS_HISTORY: ProtectionRequest[] = [
    {
        id: "1",
        requesterAddress: "5Ew3MyB15VprZrjQVkpQFj8okmc9xLDSEdNhqMMS5cXsqxoW",
        legalOfficerAddress: DEFAULT_LEGAL_OFFICER,
        otherLegalOfficerAddress: ANOTHER_LEGAL_OFFICER,
        decision: {
                rejectReason: null,
                decisionOn: toIsoString(moment('2021-06-10T11:40:00.000Z')),
                locId: "locId"
        },
        userIdentity: {
            firstName: "John",
            lastName: "Doe",
            email: "john.doe@logion.network",
            phoneNumber: "+1234",
        },
        userPostalAddress: {
            line1: "Place de le République Française, 10",
            line2: "boite 15",
            postalCode: "4000",
            city: "Liège",
            country: "Belgium",
        },
        createdOn: toIsoString(moment('2021-06-10T11:40:00.000Z')),
        isRecovery: false,
        addressToRecover: null,
        status: "ACTIVATED"
    }
];

export const RECOVERY_REQUESTS_HISTORY: ProtectionRequest[] = [
    {
        id: "1",
        requesterAddress: "5Ew3MyB15VprZrjQVkpQFj8okmc9xLDSEdNhqMMS5cXsqxoW",
        legalOfficerAddress: DEFAULT_LEGAL_OFFICER,
        otherLegalOfficerAddress: ANOTHER_LEGAL_OFFICER,
        decision: {
            rejectReason: null,
            decisionOn: toIsoString(moment('2021-06-10T11:40:00.000Z')),
            locId: "locId"
        },
        userIdentity: {
            firstName: "John",
            lastName: "Doe",
            email: "john.doe@logion.network",
            phoneNumber: "+1234",
        },
        userPostalAddress: {
            line1: "Place de le République Française, 10",
            line2: "boite 15",
            postalCode: "4000",
            city: "Liège",
            country: "Belgium",
        },
        createdOn: toIsoString(moment('2021-06-10T11:40:00.000Z')),
        isRecovery: true,
        addressToRecover: "an-address",
        status: "PENDING"
    }
];
