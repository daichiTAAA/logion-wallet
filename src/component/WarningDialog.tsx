import React from 'react';

import Dialog, { ModalSize } from './Dialog';
import { Action } from './Button';
import Icon from './Icon';

import { Children } from './types/Helpers';
import { ColorTheme } from './ColorTheme';

import './WarningDialog.css';

export interface Props {
    show: boolean,
    modalTestId?: string,
    actions: Action[],
    size: ModalSize,
    colors: ColorTheme,
    children: Children,
    spaceAbove?: string,
}

export default function WarningDialog(props: Props) {

    return (
        <Dialog
            show={ props.show }
            size={ props.size }
            actions={ props.actions }
            colors={ props.colors }
            modalTestId={ props.modalTestId }
            spaceAbove={ props.spaceAbove }
        >
            <Icon
                colorThemeType={ props.colors.type }
                icon={{
                    id: "warning"
                }}
            />
            <p className="dialog-text">
                { props.children }
            </p>
        </Dialog>
    );
}
