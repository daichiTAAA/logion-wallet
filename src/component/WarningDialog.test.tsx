import { shallowRender } from '../tests';
import WarningDialog from './WarningDialog';
import { COLOR_THEME } from './TestData';

test("renders with default space above", () => {
    const result = shallowRender(<WarningDialog
        show={ true }
        actions={[
            {
                id: "button",
                buttonText: "Text",
                callback: jest.fn(),
                buttonVariant: 'primary'
            }
        ]}
        size="lg"
        colors={ COLOR_THEME }
    >
        Content
    </WarningDialog>);
    expect(result).toMatchSnapshot();
});

test("renders with custom space above", () => {
    const result = shallowRender(<WarningDialog
        show={ true }
        actions={[
            {
                id: "button",
                buttonText: "Text",
                callback: jest.fn(),
                buttonVariant: 'primary'
            }
        ]}
        size="lg"
        colors={ COLOR_THEME }
        spaceAbove="200px"
    >
        Content
    </WarningDialog>);
    expect(result).toMatchSnapshot();
});
