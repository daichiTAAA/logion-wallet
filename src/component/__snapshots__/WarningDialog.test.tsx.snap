// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with custom space above 1`] = `
<Dialog
  actions={
    Array [
      Object {
        "buttonText": "Text",
        "buttonVariant": "primary",
        "callback": [MockFunction],
        "id": "button",
      },
    ]
  }
  colors={
    Object {
      "accounts": Object {
        "background": "eff3fe",
        "foreground": "#000000",
        "hintColor": "#00000099",
        "iconBackground": "#3b6cf4",
        "legalOfficerIcon": Object {
          "category": "legal-officer",
          "id": "account-shield",
        },
      },
      "bottomMenuItems": Object {
        "iconGradient": Object {
          "from": "#7a90cb",
          "to": "#3b6cf4",
        },
      },
      "buttons": Object {
        "secondaryBackgroundColor": "#eff3fe",
      },
      "dashboard": Object {
        "background": "#eff3fe",
        "foreground": "#000000",
      },
      "frame": Object {
        "background": "#ffffff",
        "foreground": "#000000",
        "link": "#3b6cf426",
      },
      "select": Object {
        "background": "#ffffff",
        "foreground": "#000000",
        "menuBackgroundColor": "#ffffff",
        "selectedOptionBackgroundColor": "#ffffff",
      },
      "shadowColor": "#3b6cf41a",
      "sidebar": Object {
        "activeItemBackground": "#d8e2fd",
        "background": "#ffffff",
        "foreground": "#000000",
      },
      "topMenuItems": Object {
        "iconGradient": Object {
          "from": "#3b6cf4",
          "to": "#6050dc",
        },
      },
      "type": "light",
    }
  }
  show={true}
  size="lg"
  spaceAbove="200px"
>
  <Icon
    colorThemeType="light"
    icon={
      Object {
        "id": "warning",
      }
    }
  />
  <p
    className="dialog-text"
  >
    Content
  </p>
</Dialog>
`;

exports[`renders with default space above 1`] = `
<Dialog
  actions={
    Array [
      Object {
        "buttonText": "Text",
        "buttonVariant": "primary",
        "callback": [MockFunction],
        "id": "button",
      },
    ]
  }
  colors={
    Object {
      "accounts": Object {
        "background": "eff3fe",
        "foreground": "#000000",
        "hintColor": "#00000099",
        "iconBackground": "#3b6cf4",
        "legalOfficerIcon": Object {
          "category": "legal-officer",
          "id": "account-shield",
        },
      },
      "bottomMenuItems": Object {
        "iconGradient": Object {
          "from": "#7a90cb",
          "to": "#3b6cf4",
        },
      },
      "buttons": Object {
        "secondaryBackgroundColor": "#eff3fe",
      },
      "dashboard": Object {
        "background": "#eff3fe",
        "foreground": "#000000",
      },
      "frame": Object {
        "background": "#ffffff",
        "foreground": "#000000",
        "link": "#3b6cf426",
      },
      "select": Object {
        "background": "#ffffff",
        "foreground": "#000000",
        "menuBackgroundColor": "#ffffff",
        "selectedOptionBackgroundColor": "#ffffff",
      },
      "shadowColor": "#3b6cf41a",
      "sidebar": Object {
        "activeItemBackground": "#d8e2fd",
        "background": "#ffffff",
        "foreground": "#000000",
      },
      "topMenuItems": Object {
        "iconGradient": Object {
          "from": "#3b6cf4",
          "to": "#6050dc",
        },
      },
      "type": "light",
    }
  }
  show={true}
  size="lg"
>
  <Icon
    colorThemeType="light"
    icon={
      Object {
        "id": "warning",
      }
    }
  />
  <p
    className="dialog-text"
  >
    Content
  </p>
</Dialog>
`;
